packages/client/src/client/open-feature-client.ts

```diff
@@ -225,6 +226,16 @@ export class OpenFeatureClient implements Client {
         flagKey,
       };
 
+      if (evaluationDetails.errorCode || evaluationDetails.reason === StandardResolutionReasons.ERROR) {
+        const errorCode = evaluationDetails.errorCode || ErrorCode.GENERAL;
+        this.errorHooks(allHooksReversed, hookContext, instantiateErrorByKey(errorCode), options);
+        return {
+          ...evaluationDetails,
+          errorCode,
+          reason: StandardResolutionReasons.ERROR,
+        };
```

Comment:
I'm also setting the reason to error. This will override the reason set by the provider. Does this make sense or should we return the reason defined in the evaluation details?

Reply:
I think error is good here!